src/BenchmarkDotNet/Toolchains/CsProj/CsProjGenerator.cs

```diff
@@ -133,9 +135,23 @@ protected virtual string GetRuntimeSettings(GcMode gcMode, IResolver resolver)
                 if (trimmedLine.StartsWith("<Project Sdk=\"")
                     || (TargetFrameworkMoniker.StartsWith("netcoreapp", StringComparison.InvariantCultureIgnoreCase) &&  trimmedLine.StartsWith("<Import Sdk=\"")))
                     sdkName = trimmedLine.Split('"')[1]; // its sth like Sdk="name"
+
+                if (trimmedLine.StartsWith("<PackageReference Include=\""))
```

Comment:
I think to handle all weird formatting issues would require parsing the xml, which is out of scope for this fix. I just went with the same parsing strategy that the other code lines used. I think #1403 would cover that.

Reply:
Could we at least handle unusual formatting in `AppendPackageReference`? It seems that in case of incorrect formatting, this method may go to the end of the file and lead to generation of incorrect XML (in case the closing part of the PackageReference is not at the end of the line). For such a strangely formatted csproj, the suggested PR would introduce a regression.

Reply:
How so? AppendPackageReference correctly checks for the closing tag, whether it's on the same line or a following line.

Reply:
@AndreyAkinshin I went ahead and refactored it to use xml parsing to assuage your concerns.

Reply:
@timcassell Awesome, thanks! Will review it soon.